doc: adding documentation to know how to work on the project (#165)
vuemontreal/vuemontreal: docs/architecture.md

# Architecture

> We have a classic nuxt architecture. What you should know here is our components folder:

Reviewer: Can you provide a link to some docs defining what "a classic nuxt architecture" is?
I've never worked with nuxt, so I don't know what is the standard architecture.

Author: just the basic directory structure and add the vue design system pattern for our components

https://nuxtjs.org/guide/directory-structure

## Tokens

> We are following as mush as possible [Vue design system](https://github.com/viljamis/vue-design-system/wiki/working-with-the-system).

Reviewer: Typo
```suggestion
We are following as much as possible [Vue design system](https://github.com/viljamis/vue-design-system/wiki/working-with-the-system).
```

`Design tokens are the atoms of the system as Salesforce describes them. In Vue Design System they are used instead of hard coded values to ensure a better consistency across any platform.`

You will have a lot of informations here inside our [styleguide]('./styleguide.md)

## Elements

`Elements are the smallest basic structures of a UI. They can not be broken down any further. Buttons, links, and inputs are good examples. Elements utilize Tokens.`

Creating a new Element:

> To create a new element, you will first want to navigate to /components/elements/ and create a new .vue file. Element names don’t have a prefix, but it’s recommended to make sure that they are compatible with existing and future HTML elements (to learn more, see Naming of Things).

Reviewer: ```suggestion
To create a new element, you will first want to navigate to /components/elements/ and create a new .vue file. Element names don’t have a prefix, but it’s recommended to make sure that they are compatible with existing and future HTML elements (to learn more, see [Naming of Things](https://github.com/viljamis/vue-design-system/wiki/naming-of-Things)).
```

> At the they should be inside your '/components/index.js' and exported with Ex Prefix.

Reviewer: Typo?
```suggestion
At the end they should be inside your `/components/index.js` and exported with Ex Prefix.
```

## Patterns

`Patterns are UI patterns that fall on the more complex side of the spectrum. Patterns can consist of both Elements and Tokens.`

> Example here our card event and job components.

Reviewer: Added links to elements
```suggestion
Example here our [card event](https://vuemontreal-designsystem.netlify.app/?path=/story/elements--card-event) and [job component](https://vuemontreal-designsystem.netlify.app/?path=/story/elements--card-job).
```

The VueMontreal Design System is based on [atomic web design principles](http://bradfrost.com/blog/post/atomic-web-design/).

![img atomic web design principles](./design-system.svg)

### Tokens

`Design tokens are the atoms of the system as Salesforce describes them. In Vue Design System they are used instead of hard coded values to ensure a better consistency across any platform.`

### Elements

`Elements are the smallest basic structures of a UI. They can not be broken down any further. Buttons, links, and inputs are good examples. Elements utilize Tokens.`

Element are located in [components/elements](../components/elements).  
Element names don’t have a prefix, but it’s recommended to make sure that they are compatible with existing and future HTML elements (to learn more, see [Naming of Things](https://github.com/viljamis/vue-design-system/wiki/naming-of-Things)).

Once ready, export your component in [components/index.js](../components/index.js).

### Patterns

`Patterns are UI patterns that fall on the more complex side of the spectrum. Patterns can consist of both Elements and Tokens.`

Patterns are located in [components/patterns](../components/patterns).  

> TODO: create patterns folder, and move existing one.

Reviewer: @f3ltron should we create the `patterns` folder and move components like `CardJob` and `CardEvent`?

Author: yep exactly
